eclair-core/src/test/scala/fr/acinq/eclair/io/PeerConnectionSpec.scala

```diff
@@ -372,5 +373,49 @@ class PeerConnectionSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike wi
     assert(new String(warn2.data.toArray).startsWith("invalid announcement sig"))
   }
 
+  test("establish transient connection") { f =>
+    import f._
+    connect(nodeParams, remoteNodeId, switchboard, router, connection, transport, peerConnection, peer, isPersistent = false)
+    val probe = TestProbe()
+    val (_, message) = buildMessage(randomKey(), randomKey(), Nil, Left(Recipient(remoteNodeId, None)), Nil)
+    probe.send(peerConnection, message)
+    probe watch peerConnection
+    Thread.sleep(900)
```

Comment:
We should generally avoid `Thread.sleep` unless absolutely required, because it blocks an entire thread which could otherwise be used to run concurrent tests if you used a non-blocking construct instead. This will be especially penalizing during CI builds.

In this specific test, you can replace it with:

```scala
awaitCond(peerConnection.stateName === PeerConnection.CONNECTED)
probe.expectTerminated(peerConnection, max = Duration.Zero)
```